### PR TITLE
Add bond migration

### PR DESCRIPTION
**Pull request by cfconrad**
## Problem

Upstream bonding was finally merged and now we can re-implement the migrate-wicked side.

## Testing

- *Added a new unit test*

### CODE REVIEW

**Review by jcronenberg — CHANGES_REQUESTED** [review]

File: rust/migrate-wicked/src/migrate.rs [hunk 1]
```diff
@@ -12,10 +13,40 @@ impl WickedAdapter {
...
+            }
+        }
+    }
+
+    for (id, uuid) in parent_uuid {
+        if let Some(connection) = state.get_connection_mut(&id) {
+            connection.controller = Some(uuid);
+        } else {
+            return Err(anyhow::anyhow!("Unexpected failure - missing connection {}", id).into());
+        }
```
[the remark this answers is not in the document]
Comment (jcronenberg) [reply rust/migrate-wicked/src/migrate.rs:39]: Nvm, after thinking about it a bit more, I dislike my version more and I don't see any benefit :slightly_smiling_face:

File: rust/migrate-wicked/src/interface.rs [hunk 2]
```diff
@@ -126,24 +453,32 @@ impl Interface {
...
+            }
             connection.config = model::ConnectionConfig::Dummy
+        } else if let Some(bond) = &self.bond {
+            if let Some(address) = &bond.address {
+                connection.mac_address = MacAddress::from_str(address)?;
+            }
+            connection.config = bond.into()
         } else {
             connection.config = model::ConnectionConfig::Ethernet
         };
```
[the remark this answers is not in the document]
Comment (cfconrad) [reply rust/migrate-wicked/src/interface.rs:484]: WDYT of 
```rust
impl TryFrom<&Option<String>> for MacAddress {
    type Error = InvalidMacAddress;

    fn try_from(value: &Option<String>) -> Result<Self, Self::Error> {
        match &value {
            Some(str) => MacAddress::from_str(str),
            None => Ok(Self::Unset)
        }
    }
}
```

[the remark this answers is not in the document]
Comment (jcronenberg) [reply rust/migrate-wicked/src/interface.rs:484]: Sure, should also work if you prefer it.

[the remark this answers is not in the document]
Comment (cfconrad) [reply rust/migrate-wicked/src/interface.rs:484]: I finally go with 
```
#serde(default)
address: String,
```
which lead to empty string if it isn't in xml...

[the remark this answers is not in the document]
Comment (jcronenberg) [reply rust/migrate-wicked/src/interface.rs]: > I finally go with
> 
> ```
> #serde(default)
> address: String,
> ```
> 
> which lead to empty string if it isn't in xml...

In that case you should also add
```rust
#[serde(skip_serializing_if = "String::is_empty")]
```

[the remark this answers is not in the document]
Comment (cfconrad) [reply rust/migrate-wicked/src/interface.rs]: valid point. 

Lets see if this https://github.com/openSUSE/agama/pull/983 gets accepted, maybe I go back to `Option<String>`

**Review by jcronenberg — APPROVED** [review]

**Pull request merged**